lib/DebugInfo/Symbolize/BugsnagSymbolize.cpp

```diff
@@ -0,0 +1,100 @@
+#include "llvm/DebugInfo/Symbolize/BugsnagSymbolize.h"
+#include "llvm/DebugInfo/Symbolize/Symbolize.h"
+#include <cstdlib>
+#include <cstring>
+
+using namespace llvm;
+
+static void freeAndInvalidate(void** p) {
+  free((void*)*p);
+  *p = nullptr;
+}
+
+static void destroySymbolizeResult(SymbolizeResult* symbolizeResult) {
+  if (!symbolizeResult) {
+    return;
+  }
+  freeAndInvalidate((void**)&symbolizeResult->fileName);
+  freeAndInvalidate((void**)&symbolizeResult->shortFunctionName);
+  freeAndInvalidate((void**)&symbolizeResult->linkageFunctionName);
+  freeAndInvalidate((void**)&symbolizeResult->symbolTableFunctionName);
+}
+
+static void destroySymbolizeResults(SymbolizeResults* symbolizeResults) {
+  if (!symbolizeResults) {
+    return;
+  }
+  for (int i = 0; i < symbolizeResults->resultCount; i++) {
+    destroySymbolizeResult(&symbolizeResults->results[i]);
+  }
+}
+
+SymbolizeResult getSymbolizeResult(const DILineInfo &info, int64_t address, bool inlined) {
+  SymbolizeResult result = {0};
+
+  result.address = address;
+  result.inlined = inlined;
+
+  unsigned int len = std::strlen(info.FileName.c_str());
+  result.fileName = new char[len];
+  std::strcpy(result.fileName, info.FileName.c_str());
+
+  len = std::strlen(info.ShortFunctionName.c_str());
+  result.shortFunctionName = new char[len];
+  std::strcpy(result.shortFunctionName, info.ShortFunctionName.c_str());
+
+  len = std::strlen(info.LinkageFunctionName.c_str());
+  result.linkageFunctionName = new char[len];
+  std::strcpy(result.linkageFunctionName, info.LinkageFunctionName.c_str());
+
+  len = std::strlen(info.SymbolTableFunctionName.c_str());
+  result.symbolTableFunctionName = new char[len];
+  std::strcpy(result.symbolTableFunctionName, info.SymbolTableFunctionName.c_str());
+
+  result.line = info.Line;
+  result.column = info.Column;
+  result.startLine = info.StartLine;
+
+  return result;
+}
+
+SymbolizeResults BugsnagSymbolize(const char* filePath, bool includeInline, int64_t addresses[], int addressCount) {
```

Comment:
The addresses field in the signature for symbolizeCode in Symbolize.h looks to be a uint64_t, is there a reason this uses int64_t instead?

Reply:
This was changed under [bugsnag/llvm@`bce7a26` (#3)](https://github.com/bugsnag/llvm/pull/3/commits/bce7a2690a86d2e06d82f4a1d94b68ca3646c5d8)